kong/plugins/grpc-gateway/deco.lua

```diff
@@ -78,6 +79,47 @@ local function parse_options_path(path)
   return path_regex, match_groups
 end
 
+local function safe_set_type_hook(type, dec, enc)
+  if not pcall(pb.hook, type) then
+    ngx.log(ngx.NOTICE, "no type '" .. type .. "' defined")
+    return
+  end
+
+  if not pb.hook(type) then
+    pb.hook(type, dec)
```

Comment:
Sadly, this solution cannot handle native types, namely `bytes` - https://github.com/starwing/lua-protobuf/issues/161#issuecomment-927001222

Bytes are handled in special way https://developers.google.com/protocol-buffers/docs/proto3#json

We fixed this by manually traversing the message looking for type and calling respective hook. I can push my solution, but first I want to integrate it with yours.